File: examples/python/send_text_message.py
"""
Contoh Kode: Mengirim Pesan Text

File ini berisi contoh implementasi untuk mengirim pesan text melalui WhatsApp
"""

from config import make_request, display_response


def main():
    print('=== WACloud - Send Text Message Example ===\n\n')

    # Device ID yang sudah terhubung (status: connected)
    device_id = '550e8400-e29b-41d4-a716-446655440000'

    # Nomor tujuan (format: tanpa leading 0, dengan kode negara)
    # Contoh: 6281234567890 (Indonesia)
    to = '6281234567890'

    # Data pesan
    message_data = {
        'device_id': device_id,
        'to': to,
        'message_type': 'text',
        'text': 'Halo, ini pesan dari API WACloud!'
    }

    print('Mengirim pesan text...')
    print(f'To: {to}')
    print(f"Message: {message_data['text']}\n")

    response = make_request('POST', '/messages', message_data)
    display_response(response)

    if response.get('success') and response.get('data', {}).get('message_id'):
        print('✓ Pesan berhasil dikirim!')
        print(f"Message ID: {response['data']['message_id']}")
        print(f"WhatsApp Message ID: {response['data']['whatsapp_message_id']}")
        print(f"Status: {response['data']['status']}")
    else:
        print('✗ Gagal mengirim pesan.')
        if response.get('error'):
            print(f"Error: {response['error']}")
        if response.get('message'):
            print(f"Message: {response['message']}")

    print('\n=== Selesai ===')


if __name__ == '__main__':
    main()

File: examples/nodejs/send-text-message.js
/**
 * Contoh Kode: Mengirim Pesan Text
 * 
 * File ini berisi contoh implementasi untuk mengirim pesan text melalui WhatsApp
 */

const { makeRequest, displayResponse } = require('./config');

async function main() {
    console.log('=== WACloud - Send Text Message Example ===\n\n');

    // Device ID yang sudah terhubung (status: connected)
    const deviceId = '550e8400-e29b-41d4-a716-446655440000';

    // Nomor tujuan (format: tanpa leading 0, dengan kode negara)
    // Contoh: 6281234567890 (Indonesia)
    const to = '6281234567890';

    // Data pesan
    const messageData = {
        device_id: deviceId,
        to: to,
        message_type: 'text',
        text: 'Halo, ini pesan dari API WACloud!'
    };

    console.log('Mengirim pesan text...');
    console.log(`To: ${to}`);
    console.log(`Message: ${messageData.text}\n`);

    const response = await makeRequest('POST', '/messages', messageData);
    displayResponse(response);

    if (response.success && response.data?.message_id) {
        console.log('✓ Pesan berhasil dikirim!');
        console.log(`Message ID: ${response.data.message_id}`);
        console.log(`WhatsApp Message ID: ${response.data.whatsapp_message_id}`);
        console.log(`Status: ${response.data.status}`);
    } else {
        console.log('✗ Gagal mengirim pesan.');
        if (response.error) {
            console.log(`Error: ${response.error}`);
        }
        if (response.message) {
            console.log(`Message: ${response.message}`);
        }
    }

    console.log('\n=== Selesai ===');
}

main().catch(console.error);

